modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ServiceCatalogRestAPITestCase.java

```diff
@@ -0,0 +1,511 @@
+package org.wso2.am.integration.tests.restapi;
+
+import org.apache.http.HttpStatus;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.*;
+import org.wso2.am.integration.clients.service.catalog.api.ApiException;
+import org.wso2.am.integration.clients.service.catalog.api.ApiResponse;
+import org.wso2.am.integration.clients.service.catalog.api.v1.dto.APIListDTO;
+import org.wso2.am.integration.clients.service.catalog.api.v1.dto.ServiceDTO;
+import org.wso2.am.integration.clients.service.catalog.api.v1.dto.ServiceInfoListDTO;
+import org.wso2.am.integration.clients.service.catalog.api.v1.dto.ServiceListDTO;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationBaseTest;
+import org.wso2.am.integration.test.utils.generic.TestConfigurationProvider;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+@SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+public class ServiceCatalogRestAPITestCase extends APIMIntegrationBaseTest {
+
+    private File definitionFileSampleOne;
+
+    private String serviceIdOne = "";
+
+    private String serviceIdTwo = "";
+
+    private String importedServiceId = "";
+
+    private final String invalidServiceId = "01234567-0123-0123-0123";
+
+    private final String emptyServiceId = null;
+
+    private String apiId = "";
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public ServiceCatalogRestAPITestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN},
+                new Object[]{TestUserMode.SUPER_TENANT_USER_STORE_USER},
+                new Object[]{TestUserMode.SUPER_TENANT_EMAIL_USER},
+                new Object[]{TestUserMode.TENANT_EMAIL_USER},
```

Comment:
shall we add secondary userstore user as well?

Reply:
@Sachini-Dissanayaka, as per offline discussion, lets address it in a separate PR.